server/protocols/webadmin/webadmin-mailbox/src/main/java/org/apache/james/webadmin/service/DomainQuotaService.java

```diff
@@ -84,9 +82,20 @@ public QuotaDomainDTO getQuota(Domain domain) throws MailboxException {
     }
 
     public void defineQuota(Domain domain, QuotaDTO quota) {
-        quota.getCount()
-            .ifPresent(Throwing.consumer(count -> maxQuotaManager.setDomainMaxMessage(domain, count)));
-        quota.getSize()
-            .ifPresent(Throwing.consumer(size -> maxQuotaManager.setDomainMaxStorage(domain, size)));
+        try {
+            if (quota.getCount().isPresent()) {
+                maxQuotaManager.setDomainMaxMessage(domain, quota.getCount().get());
+            } else {
+                maxQuotaManager.removeDomainMaxMessage(domain);
+            }
+
+            if (quota.getSize().isPresent()) {
+                maxQuotaManager.setDomainMaxStorage(domain, quota.getSize().get());
```

Comment:
I find it sad to not rely on optionals for this...

Reply:
Me too but the `ifPresentOrElse`  method on Optional is available only since JDK 9. So it will break compatibility with JDK8 to use it.

Reply:
```
quota.getSize()
    .map(size -> () -> maxQuotaManager.setDomainMaxStorage(domain, size))
    .orElse(() -> maxQuotaManager.removeDomainMaxStorage(domain))
    .run();
```

?

Reply:
The type inference of java doesn't work as well in real life :/ 
So I still prefer the current version.